CIP-????/README.md

```diff
@@ -0,0 +1,334 @@
+Within validation zones we can therefore allow features that require “earlier” transactions to depend on “later” transactions for their validity.
+
+At the ledger level, a validation zone:
+
+- Is a new top-level element (an alternative to a transaction)
+- Contains a sequence of transactions (not zones, there is no point nesting them)
+- Carries no further data beyond this
+- Triggers a special mode of validation for the transactions within (see “validating a zone” below)
+
+### Unresolved inputs and resolving outputs
+
+We add a new kind of transaction input, an “unresolved input” (UTXI). 
+An unresolved input differs from a normal transaction input:
+
+- It does not specify a transaction output that it spends
+- It does specify the value that the input provides
+- It has a credential and an optional datum
+
+Unresolved inputs must be addressable. 
+Current transaction inputs are not addressable by the combination of transaction ID and an index, since they are kept in a set. 
+We therefore need to implement https://github.com/cardano-foundation/CIPs/pull/758, or to keep the unresolved inputs in a separate list.
+
+We add a new kind of transaction output, a “resolving output”. 
+A resolving output differs from a normal transaction output:
+- It specifies that it “resolves” a particular unresolved transaction input, by specifying the transaction ID and index of the input
+- It does not specify the value that it provides
+- It does not have a credential
+
+Unresolved inputs and resolving outputs are illegal outside a validation zone. 
+They are included in the script context accurately, with one subtlety: in the same way that normal transaction inputs are decorated with the value that they provide, resolving transaction outputs are similarly decorated with the value that they provide.
+
+### Required transactions
+
+We add a new field to the transaction body, `requiredTxs`, which is a list of transaction IDs.
+
+The `requiredTxs` field is special in that it is _not_ included when hashing the transaction body to create the transaction ID. 
+It is however signed when the transaction body is signed.
+
+The required transactions field is illegal outside a validation zone. 
+It is included in the script context.
+
+### Validating a zone
+
+Validating a zone proceeds as follows:
+
+#### 1: Check all zone-level properties
+The following zone-level properties are checked:
+
+1. (UTXIs are resolved) The unresolved inputs and resolving outputs in the zone match up exactly.
+2. (Linearizability) The transactions are linearizable, that is: there is a reordering of the transactions in the zone such that every resolving output comes before every unresolved input which it resolves.
+3. (Required transactions) For every transaction, all the transactions listed in `requiredTxs` are present in the zone.
+
+If any of these fail, the entire zone is discarded.
```

Comment:
This is conflating ledger checks with what a mempool would do, i think? A mempool would discard a zone, which a ledger state update would fail to apply the block containing the zone

Reply:
Right, I guess I want to just say "invalid"? And then the action depends on who is doing the checking.